hy/compiler.py

```diff
@@ -1139,6 +1141,14 @@ def compile_yield_expression(self, expr):
         node = asty.Yield if expr[0] == "yield" else asty.YieldFrom
         return ret + node(expr, value=ret.force_expr)
 
+    @builds("await", iff=PY35)
+    @checkargs(max=1)
```

Comment:
`(await)` should be illegal, since `await` needs an expression in Python.

Reply:
How do I do that? I based this block on `yield`/`yield-from`, so wouldn't they share that problem then?

Reply:
Write `@checkargs(1)` instead of `@checkargs(max=1)` and simplify the method body accordingly. In Python, nullary `yield` is equivalent to `yield None`, so `(yield)` should work similarly. On the other hand, the fact that `(yield-from)` is allowed is apparently a bug; `yield from None` will never work.

Reply:
Maybe I should adjust this so that `yield` is its own block with `await` and `yield from` sharing their implementation instead?

Reply:
That would be fine, I think.